File: ruby/rna-transcription/rna_transcription.rb
# frozen_string_literal: true

module Complement
  DNA_COMPLEMENT = { 'G' => 'C',
                     'C' => 'G',
                     'T' => 'A',
                     'A' => 'U' }.freeze
  def self.of_dna(strand)
    strand.chars.map { |nucleotide| DNA_COMPLEMENT[nucleotide] }.join
  end
end

File: python/rna-transcription/rna_transcription.py
RNA_MAPPING =   {'G' : 'C',
                'C' : 'G',
                'T' : 'A',
                'A' : 'U'}

def to_rna(dna_strand):
  return ''.join([ RNA_MAPPING[nucleotide]  for nucleotide in dna_strand])
